### ISSUE TITLE
Configurable character limit on regex triggers

### ISSUE THREAD

**Issue by AucaCoyan** [espanso/espanso#1766]
### Discussed in https://github.com/espanso/espanso/discussions/1253

<div type='discussions-op-text'>

<sup>Originally posted by **Staples1010** January 23, 2022</sup>
I would like to use larger character counts in my arguments, I'm a web dev and would like to use it to create anchors. For example:
```<a href="{{arg1}}" target="_blank">{{arg2}}</a>```
links and anchors will pretty much always be over 30 characters :/

Could it be configurable in the default.yml file?</div>

Hi! Just transferring this discussion into an Issue. First contributors more than welcome!

**Comment by joh6nn** [comment]
I doubt there's a way to accomplish this that doesn't impact performance. I think if people need longer arguments the best option is to use the forms extension

**Comment by AucaCoyan** [comment]
Yeah, that's true. At least we could test some increased limits and see how much longer it takes.

**Comment by smeech** [comment]
No harm in having it as an *option* with appropriate warnings regarding performance.

**Comment by gruns** [comment]
just ran into this issue myself: `replace` strings longer than 103 characters in length fail

103 chars succeeds:
```
- triggers: ['@@grep']
  replace: |
    grep -riI --exclude-dir=node_modules --exclude-dir=dist --exclude-dir=.to --exclude='*.min.js*' '$|$' .
      
>>> len("grep -riI --exclude-dir=node_modules --exclude-dir=dist --exclude-dir=.to --exclude='*.min.js*' '$|$' .")
103
```

104 chars fails:
```
- triggers: ['@@grep']
  replace: |
    grep -riI --exclude-dir=node_modules --exclude-dir=dist --exclude-dir=.tox --exclude='*.min.js*' '$|$' .
      
>>> len("grep -riI --exclude-dir=node_modules --exclude-dir=dist --exclude-dir=.tox --exclude='*.min.js*' '$|$' .")
104
```

`use_simulator: true` doesnt work as a workaround, unfortunately

is there a standard workaround here?

**Comment by AucaCoyan** [comment]
Not that I know yet. I think we need to change the code to be able to do longer regexes.
Of course it's necessary that this would be hidden it under a custom option `regex-length = 300` or something like that.

**Comment by smeech** [comment]
> just ran into this issue myself: `replace` strings longer than 103 characters in length fail
> 
> 103 chars succeeds:
> 
> ```
> - triggers: ['@@grep']
>   replace: |
>     grep -riI --exclude-dir=node_modules --exclude-dir=dist --exclude-dir=.to --exclude='*.min.js*' '$|$' .
>       
> >>> len("grep -riI --exclude-dir=node_modules --exclude-dir=dist --exclude-dir=.to --exclude='*.min.js*' '$|$' .")
> 103
> ```
> 
> 104 chars fails:
> 
> ```
> - triggers: ['@@grep']
>   replace: |
>     grep -riI --exclude-dir=node_modules --exclude-dir=dist --exclude-dir=.tox --exclude='*.min.js*' '$|$' .
>       
> >>> len("grep -riI --exclude-dir=node_modules --exclude-dir=dist --exclude-dir=.tox --exclude='*.min.js*' '$|$' .")
> 104
> ```
> 
> `use_simulator: true` doesnt work as a workaround, unfortunately
> 
> is there a standard workaround here?

This particular problem probably isn't related to the `regex:` Issue, above.

Injection failures around this length are usually due to the `espanso/config/default.yml` `backend: auto` changeover from `inject` to `clipboard` that occurs at the 100 character default value for [`clipboard_threshold:`](https://espanso.org/docs/configuration/options/) in the same file (which doesn't count the `$|$`). Try increasing this a little to see if it makes a difference, but you may need an [app-specific configuration](https://espanso.org/docs/configuration/app-specific-configurations/). With correct configurations, I've tested `replace:` strings several thousand characters long, so we should be able to fix your problem.

**Comment by aldair-meneses** [comment]
> Not that I know yet. I think we need to change the code to be able to do longer regexes. Of course it's necessary that this would be hidden it under a custom option `regex-length = 300` or something like that.

I made this [PR](#2504) because, in my case, when I was integrating with an AI and didn’t want to use a form, I needed to modify the code. This solution works for me, so I thought it would be great to have it built in instead of requiring a patch.

**Comment by AucaCoyan** [comment]
Indeed! I think is a good solution, though I didn't had time to compile and review it in detail.
I'm almost there!
